### DISCUSSION TITLE
Rev V2 Suggestion Tracking

### DISCUSSION THREAD

**Discussion by Dark1886** [OMOTE-Community/OMOTE-Firmware#24] [Ideas]
If you feel there is something with the current board/Component selection that could be addressed in a very hypothetical V2 board, leave a comment below and we will gather a list of Pros/Cons for possible updates. 

### Update List
**MCU Change**
Possible another module with more GPIO such as [ESP32-S3-MINI-1U](https://www.espressif.com/sites/default/files/documentation/esp32-s3-mini-1_mini-1u_datasheet_en.pdf) Either with PCB antenna or uFL antenna

_Pros_
1. Higher GPIO Count
2. More Processing Power, Dual core. (I believe the pico-d4 is single core, but Confirmation may be needed)

_Cons_
1. Power consumption is likely higher on this module. Although dual core and clock speed could be tuned to match user performance/battery life concerns. 
2. Size, This module it quite a bit bigger, It may cause placement issues on the board or interfere with others. 

_________________________________________________________

**Additional Device to handle Keypad (GPIO expander/Controller)**

_Pros_
1. Using an SPI/I2C device for this could open up multiple other GPIO lanes for other functions, removing the need for many connections directly from the keypad to the MCU

_Cons_
1.  Added software complexity, possible communication issues.
2. Possibly added processing power required. 

_________________________________________________________

**Move Form SPI display control to 8-bit parallel**

_Pros_
1.  Likely much faster refresh rate or same refresh rate with fewer SPI problems.

_Cons_
1. Complexity
2. Additional GPIO required

_________________________________________________________
**Component/component size changes**

_Pros_
1. Using larger components can make it easier to solder the board by hand
2. Easier signal probing if problems arise.

_Cons_
1. More board space
2. Some components may not be available in larger sizes, alternates will be needed.

_________________________________________________________
**Adding Test Points**

_Pros_
1. Safer/easier probing.
2. Board modifications can be easier if you have test points to solder to.

_Cons_
1. Board Space
2. Possible signal integrity problems.

_________________________________________________________
Adding additional capacitor across the ENABLE pin of the MCU and GND.

**Change Item**

_Pros_
1. Should assist with minimizing "Invalid Head of Packet" errors when flashing the MCU

_Cons_
1. N/A
_________________________________________________________
Template For later use

**Change Item**

_Pros_
1. 

_Cons_
1.

**Comment by CoretechR** [comment]
Good idea to collect these ideas here!
The ESP32-PICO-V3 is the new version of the ESP32-PICO-D4 and integrates PSRAM. This is something the project could benefit from and it would be easy to integrate.
A parallel interface instead of SPI for the display would help a lot. It might be possible with some kind of IO expander for the keypad to free up those pins.

**Reply by Dark1886** [reply]
I figured I wanted to gather all my thoughts in one place from what I see. Since I work in electronics functional testing for a living, I figured I could also have some unique experience when it comes to product debugging and troubleshooting. Plus a fresh set of eyes on a problem always beings a new perspective. 

As for the parallel display/Keypad IO expander trade off, it may be something I look into later after getting my current controller working as expected.

**Reply by CoretechR** [reply]
> Since I work in electronics functional testing for a living...

The request for test points gave you away. 😄
I really appreciate the feedback and ideas. It's tempting to create a new version with a keypad IC and parallel display interface. That would solve a lot of problems. But many people already ordered their boards, so support for the current version is my priority.

**Reply by phant0mias** [reply]
@CoretechR: Yes you are right. Changing the hardware so fast would be not a very good idea. E.G. we ordered 10 boards and start building them together, creating cases and start implementing the software.
There are so many things in software that can be done before like:
- Weatherreport
- Amazon Fire Stick compatibility
- TV program display
Thanks for your work

**Reply by samdowrick** [reply]
Already working on fire stick compatibility using adb commands.

**Reply by phant0mias** [reply]
that is a good idea, i tried using bluetooth, but had no success.

**Reply by samdowrick** [reply]
It's proving quite difficult as there are no adb command libraries that are compatible

**Reply by phant0mias** [reply]
If I am right Homematic is using adb commands. This theme is new for me, I will google tonight how it is working...

**Reply by phant0mias** [reply]
also found no library...

**Reply by phant0mias** [reply]
have a look to:
https://github.com/CoretechR/OMOTE/discussions/48

**Comment by CoretechR** [comment]
Just referencing @samdowrick's brilliant idea for using a 555 timer  to limit the IR LED current: https://github.com/CoretechR/OMOTE/issues/32#issuecomment-1673976016
This could save the LED in case of a software bug that powers it on indefinitely.

**Reply by Dark1886** [reply]
I like this idea a lot. I think we should have plenty of board space to add such an item if we just shift a few things around.

**Reply by samdowrick** [reply]
More than happy to contribute to any hardware changes. I'm an electrical r&d engineer for a capitor manufacturer and mostly focus on high power rf circuits however I also work in applications and spend a lot of time working with a massive range of circuits from light switches through to satelite com systems.

**Comment by alexiri** [comment]
The USB-C port is fine, but it would be great to also have a couple of "charge pads" at the bottom so we could create a cradle for it, like the Harmony Elite.

<img src="https://user-images.githubusercontent.com/5055697/268041454-ebfcf397-aafa-4124-94d3-f3e837d3228b.png" alt="image" width="650" height="587">

**Reply by Dark1886** [reply]
That would be cool. I wonder if something similar to this would work. 

https://www.adafruit.com/product/5360
![image](https://github.com/CoretechR/OMOTE/assets/64846241/25b0fc1a-b716-4eaf-883a-5138a40225da)

**Reply by Dark1886** [reply]
For the time being,you may be able to make a dock for the current remotr using a magnetic breakawau cable like this on too. 

https://www.adafruit.com/product/5520
![image](https://github.com/CoretechR/OMOTE/assets/64846241/b860661b-0b22-4d20-ae6e-86ca748b5a0e)

**Reply by WizardCM** [reply]
I can confirm that the current 3D printed case happily fits [this magnetic connector](https://www.amazon.com.au/BIGCOON-Rotating-Magnetic-Charging-Compatible/dp/B0B3W79941/ref=pd_bxgy_d_sccl_1/355-9509377-2744638?th=1). Just need some kind of cradle.

![image](https://github.com/CoretechR/OMOTE/assets/941350/0f612b14-c924-42dc-ab01-82ee034faad5)

**Comment by DJAlPee** [comment]
Has someone already thought of "re-using" existing remotes (no-names or even Harmony)?
Would be awesome to provide PCB designs, that are in best "case" (pun not intended 😄) a drop-in replacement for the existing ones.

The re-use (or replacement) of built-in displays might be the biggest challenge...

**Reply by Dark1886** [reply]
I can't say I have thought of this, but I suspect it would be a big hassle depending on the internals of the boards. Maybe it's possible it could happen in the future, but the original idea for this project was a fully custom remote so it continued that way. Adding additional design constraints at this point (to re-use components from something else) may just make it even more complicated. 

That being said, it's not an impossible task in the future.

**Reply by DJAlPee** [reply]
Makes totally sense! Thx for the fast feedback and in general for your project! 😉

**Reply by Dark1886** [reply]
I'm giving the project credit to @CoretechR, he is the original maker for this remote. Others have just joined recently to help increase functionality and make it easier for other people to customize it.

**Comment by phant0mias** [comment]
> **MCU Change** Possible another module with more GPIO such as [ESP32-S3-MINI-1U](https://www.espressif.com/sites/default/files/documentation/esp32-s3-mini-1_mini-1u_datasheet_en.pdf) Either with PCB antenna or uFL antenna
> 
> _Pros_
> 
> 1. Higher GPIO Count
> 2. More Processing Power, Dual core. (I believe the pico-d4 is single core, but Confirmation may be needed)

The ESP32 Pico D4 is a dual core MCU. In the espressif data sheet is written:
**ESP32 contains two low-power Xtensa® 32-bit LX6 microprocessors.**
(https://www.espressif.com/sites/default/files/documentation/esp32-pico-d4_datasheet_en.pdf)

It is working I already use the second core.

**Comment by ScanuNicco** [comment]
In the future it might be nice to add a small speaker or buzzer to make it easier to find a lost remote.

**Reply by Dark1886** [reply]
Hmm I like this idea, Hopefully we can see it in the next rev.

**Reply by phant0mias** [reply]
I think that this feature will not work for the remote control. As you can see in the current implementation of the software the remote control goes into deep sleep mode after 20 seconds. In deep sleep the most of the controller (including all communcation) is down, the energy eating parts are off. That is the reason why the remote can run other months. You will have no possibilty to communicate and search for it. If you are not using the deep sleep, the battery will get faster empty. I don´t know how fast, but you can check it if you remove deep sleep part in the software. And if the is battery empty, searching will not work too and what is anoying, you have to load your remote very often.

A second thing is you have to write a software on another device to search for the remote.

**Reply by CoretechR** [reply]
@phant0mias is right. This would not work with the current power management. Are there other use cases for a buzzer or speaker?

**Comment by CoretechR** [comment]
As mentioned in issue #53, some kind of over discharge protection for the battery would be useful. The current design relies on the integrated protection of the battery.

**Reply by ScanuNicco** [reply]
It would be nice to see the documentation updated to tell people that they have to buy a battery with over-discharge protection. I almost ended up buying some that didn't have it.

**Reply by MiniJe** [reply]
https://www.ti.com/power-management/battery-management/protectors/products.html#404=Protection&1152max=1%3B1&

**Reply by MiniJe** [reply]
https://www.tme.eu/ro/details/accu-lp503448_cl/acumulatori/cellevia-batteries/lp503448/

**Comment by dekay** [comment]
I have a number of suggestions
- use multiple IR emitters to increase the field of view to multiple devices and reduce the need for IR Blasters
- recommend a different IR emitter in the base BOM and resize the series resistor appropriately if required.  I believe [the current one](https://www.mouser.ca/ProductDetail/ams-OSRAM/SFH-4346?qs=cd3lnLqJ6yUhOEyi4jW7OA%3D%3D) is obsolete and has a very narrow beamwidth.  Something recommended for new designs, has a wider beamwidth, and is still very bright should be selected e.g. [SFH4547](https://www.mouser.ca/ProductDetail/720-SFH4547)?
- Consider some kind of way to backlight the keys.  My Logitech 650 does this and it is very handy in a dark room.  The 650 has pretty slick keys made of separate plastics for the body of the key and the legend, the latter which lets some light to pass through.  We couldn't easily play a trick like this on home 3D printers but there might be other ways (print buttons in transparent plastic? depend on light getting out of the spaces between the keys?)  The backlight LEDs would not need to be stuffed if the user wasn't interested in this feature.

**Reply by disconn3ct** [reply]
You can print decent LED diffusers from 1-3 layers (0.2mm to 0.6mm) of white plastic. A small CCD strip inside the case could light the internal void and shine through hollow keycaps. (Diffuser example: https://www.thingiverse.com/thing:4127790)

**Comment by FBrand** [comment]
Cheers,
I too would like to see backlight and something like pogo pins, magnets and a dock for charging.
Also an Acceleration sensor might be interesting for gestures, motion activation, etc.
Last but not least, is there any chance, to integrate the option for expansion modules into omote? That way users could add features/capabilities as needed, e.g. a rf transmitter. 
Great project, btw.
